ng/wasm/src/bls.rs

```diff
@@ -0,0 +1,201 @@
+use std::convert::TryInto as _;
+
+use bls12_381_plus::{G1Projective, G2Projective};
+use blsful::{
+    Bls12381G1Impl, Bls12381G2Impl, BlsSignatureImpl, InnerPointShareG1, InnerPointShareG2,
+    PublicKey, Signature, SignatureSchemes, TimeCryptCiphertext,
+};
+use elliptic_curve::group::GroupEncoding;
+use js_sys::Uint8Array;
+use serde::Deserialize;
+use serde_bytes::Bytes;
+use tsify::Tsify;
+use wasm_bindgen::prelude::*;
+
+use crate::abi::{from_js, into_js, JsResult};
+
+#[derive(Tsify, Deserialize)]
+#[tsify(from_wasm_abi)]
+pub enum BlsVariant {
+    Bls12381G1,
+    Bls12381G2,
+}
+
+struct Bls<C>(C);
+
+// We have to create this trait because G1/G2 implementations do not use a common trait for all the methods we need.
```

Comment:
They actually do. See https://github.com/mikelodder7/blsful/blob/main/src/traits/serdes.rs

Reply:
Not sure if we can go `Uint8Array` -> `Vec<u8>` -> `{struct}` via serde and these traits. A `TryFrom<Vec<u8>>` would fit better, or we could change the overall approach to rely on serde for JS communication everywhere.
Happy to discuss this today!